docs/dev/recipe-client-addon/workflow.rst

```diff
@@ -0,0 +1,91 @@
+Workflow
+========
+The following is a list of things you'll probably need to do at some point while
+working on the recipe client. The majority of these steps assume that you have
+a working :ref:`mozilla-central checkout <build-moz-central>`.
+
+Building an XPI
+---------------
+The ``make-xpi.sh`` script builds an XPI file from the current recipe-client
+files:
+
+.. code-block:: bash
+
+   cd recipe-client-addon
+   ./bin/make-xpi.sh # Creates /recipe-client-addon/shield-recipe-client.xpi
+
+.. _recipe-client-sync:
+
+Sync changes to mozilla-central
+-------------------------------
+The ``update-mozilla-central.sh`` script copies the recipe-client files to a
+checkout of mozilla-central. The files that are copied over are determined by
+``build-includes.txt``.
+
+.. code-block:: bash
+
+   cd recipe-client-addon
+   ./bin/update-mozilla-central.sh /path/to/mozilla-central
+
+.. _recipe-client-tests:
+
+Writing and Running Tests
+-------------------------
+The recipe client tests are designed to be run using mozilla-central's test
+frameworks. There are multiple test suites:
+
+* xpcshell_ tests, located in ``/recipe-client-addon/test/unit``, are run in a
```

Comment:
Are we going to ditch xpcshell tests? Or are we going to keep having both? If we are going to ditch them, we should make that clear here.